### PR TITLE
refactor(server): Break Apart Nexus Lambda into Smaller Parts

### PR DESCRIPTION
**Pull request by znewton**
## Description

Reviving #17529 by @zhangxin511 .

In addition to Xin's changes and fixing merge conflicts there (see first 3 commits), I also split apart Nexus' `index.ts` into smaller files. The primary change here is putting all the relevant params from `configureWebSocketServices` into 1 passed around object param (type `INexusLambdaSettings`), and moving the per-connection common variables into 1 passed around object param (type `INexusLambdaConnection`).

## Breaking Changes

Carried over from #17529:

- An invalid JWT token will now return a 401 error instead of a 500

## Reviewer Guidance

I'd recommend reviewing up to commit `ba83199` (Xin's work which includes some improvements), _then_ reviewing commit `f5849c6` (my work which should just be moving things around).

### CODE REVIEW

File: server/routerlicious/packages/lambdas/src/nexus/connect.ts [hunk 1]
```diff
@@ -0,0 +1,604 @@
...
+	}
+}
+
+async function joinRoomAndSubscribeToChannel(
+	socket: IWebSocket,
+	tenantId: string,
+	documentId: string,
+	{ logger }: INexusLambdaSettings,
+): Promise<[string, IRoom]> {
+	const clientId = generateClientId();
```
Comment (zhangxin511) [inline server/routerlicious/packages/lambdas/src/nexus/connect.ts:278]: I think for each of this stage, I added a state to record where the connection state stoppped/failed originally. Do we still want to do this?

File: server/routerlicious/packages/lambdas/src/nexus/index.ts [hunk 2]
```diff
@@ -233,7 +104,31 @@ export function configureWebSocketServices(
 	collaborationSessionEventEmitter?: TypedEventEmitter<ICollaborationSessionEvents>,
 	clusterDrainingChecker?: core.IClusterDrainingChecker,
 ) {
+	const lambdaSettings: INexusLambdaSettings = {
+		ordererManager,
```
Comment (alteut) [inline server/routerlicious/packages/lambdas/src/nexus/index.ts:108]: nit: For future maintainability, maybe we could group actual setting values in INexusLambdaSettings and then for the manager references (ordererManager,clientManager, socketTracker, etc...) have an INexusLambdaDependencies or Services. Just thought it would make things easier to track in the future

Comment (znewton) [reply server/routerlicious/packages/lambdas/src/nexus/index.ts:108]: Good idea!

File: server/routerlicious/packages/lambdas/src/nexus/connect.ts (hunk 1 above)
Comment (znewton) [reply server/routerlicious/packages/lambdas/src/nexus/connect.ts:278]: Those are down in `connectDocument` method where these are called: https://github.com/microsoft/FluidFramework/pull/19906/files/90836d9394f3dac1e97a7ac4dd69c772dac6bd42#diff-043c8ac2e224ef032226c7191a6e23f95b2139f65a140236dd2872667be1a115R488

File: server/routerlicious/packages/lambdas/src/nexus/index.ts [hunk 3]
```diff
@@ -802,41 +175,24 @@ export function configureWebSocketServices(
...
 				{ correlationId, ...baseLumberjackProperties },
 				async () =>
-					connectDocument(connectionMessage)
+					connectDocument(
+						socket,
+						lambdaSettings,
+						lambdaConnection,
+						connectionMessage,
+						properties,
+					)
```
[the remark this answers is not in the document]
Comment (znewton) [reply server/routerlicious/packages/lambdas/src/nexus/index.ts:193]: IMO these are fine because each is a strongly typed object, except for the lumber properties, so it is clear which param is which. I don't want to change all of the params of all the functions to be named from positional in this PR.

File: server/routerlicious/packages/lambdas/src/nexus/protocol.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		for (const protocolVersion of ProtocolVersions) {
+			if (semver.intersects(protocolVersion, connectVersion)) {
+				return protocolVersion;
+			}
+		}
+	}
+	return undefined;
+}
+
+export function checkVersion(versions: string[]): [string[], string] {
```
Comment (alteut) [inline server/routerlicious/packages/lambdas/src/nexus/protocol.ts:22]: nit: checkProtocolVersion,

 thinking of someone reading the connect.ts and immediately know what this function does

File: server/routerlicious/packages/lambdas/src/nexus/connect.ts [hunk 5]
```diff
@@ -79,7 +79,7 @@ function composeConnectedMessage(
 	const connectedMessage: IConnected = {
 		claims,
 		clientId,
-		existing: false,
+		existing: true,
```
Comment (zhangxin511) [inline server/routerlicious/packages/lambdas/src/nexus/connect.ts:84]: Why this change? Seems default value changed and just add a double safety confirmation

Comment (znewton) [reply server/routerlicious/packages/lambdas/src/nexus/connect.ts:84]: `true` is the value in main
https://github.com/microsoft/FluidFramework/blob/main/server/routerlicious/packages/lambdas/src/nexus/index.ts#L597

**Review by zhangxin511 — APPROVED** [review]
